### PR TITLE
Good bye, Rabbit

### PR DESCRIPTION
**Pull request by svenfuchs**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -39,4 +43,5 @@ group :test do
   gem 'database_cleaner'
   gem 'factory_girl'
   gem 'webmock'
+  gem 'sinatra-contrib'
```
Comment (meatballhat) [inline Gemfile]: Any idea why this is in the `:test` group? 🤔

File: lib/travis/hub/api/auth.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          return @app.call(env)
+        end
+
+        unauthorized
+      end
+
+      private
+
+      def challenge
+        'Basic realm="job-board"'
```
Comment (meatballhat) [inline lib/travis/hub/api/auth.rb]: Or maybe `realm="travis-hub"` 😁 ?

File: lib/travis/hub/api/auth.rb [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      end
+
+      def basic_valid?(auth)
+        user, password = auth.basic_credentials
+        return unless user && password
+        tokens[user.to_sym] == password
+      end
+
+      def bearer_valid?(auth)
+        return false if auth.job_id.nil?
```
Comment (BanzaiMan) [inline lib/travis/hub/api/auth.rb]: Is this code more succinct and as easy to read?

```ruby
!auth.job_id.nil? && !auth.params.empty?
```

File: lib/travis/hub/api/auth.rb [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+          return @app.call(env)
+        end
+
+        unless decode_jwt(auth)
+          return [403, { 'Content-Type' => 'text/plain', 'Content-Length' => '0' }, []]
+        end
+
+        if bearer_valid?(auth)
+          # env['jwt.header'] = auth.jwt_header
+          # env['jwt.payload'] = auth.jwt_payload
```
Comment (igorwwwwwwwwwwwwwwwwwwww) [inline lib/travis/hub/api/auth.rb]: what is the deal with these commented out lines?

File: lib/travis/hub/api/sentry.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+require 'raven'
+
+module Travis
+  module Hub
+    class Sentry < Sinatra::Base
+      configure do
+        Raven.configure { |c| c.tags = { environment: environment } }
```
Comment (igorwwwwwwwwwwwwwwwwwwww) [inline lib/travis/hub/api/sentry.rb]: Would be really nice to also have `x_request_id: env['HTTP_X_REQUEST_ID'])`, the way we do in API.

Comment (igorwwwwwwwwwwwwwwwwwwww) [inline lib/travis/hub/api/sentry.rb]: And perhaps also the equivalent of https://github.com/travis-ci/travis-api/blob/484e5baee3fbc872e6461842b6100e9d57b1a6f1/lib/travis/api/app/middleware/request_id.rb, that can wait until after this PR though. :)

File: Gemfile (hunk 1 above)
Comment (svenfuchs) [inline Gemfile]: @meatballhat i think the only thing we use from `contrib` is `sinatra/test-helpers`:

```
travis-hub $ ack sinatra/
lib/travis/hub/api.rb
4:require 'sinatra/base'

spec/spec_helper.rb
56:require 'sinatra/test_helpers'
```

Comment (meatballhat) [inline Gemfile]: Aha!  🎉

File: lib/travis/hub/api/sentry.rb (hunk 5 above)
Comment (svenfuchs) [inline lib/travis/hub/api/sentry.rb]: Ok if we go forward and merge this? Would welcome a PR for this :)

File: lib/travis/hub/api/auth.rb (hunk 4 above)
Comment (svenfuchs) [inline lib/travis/hub/api/auth.rb]: thanks for the catch! that's left over stuff from a copy/paste job, i think. i've removed them.

File: lib/travis/hub/api/auth.rb (hunk 3 above)
Comment (svenfuchs) [inline lib/travis/hub/api/auth.rb]: @BanzaiMan i'd normally agree, but there's now that other (previously commented out, now commented back in) condition that would make it a very long line, so i think i'll keep the current style?

File: lib/travis/hub/api/auth.rb (hunk 2 above)
Comment (svenfuchs) [inline lib/travis/hub/api/auth.rb]: good point. i've changed it to `travis-hub`.

**Pull request merged**